### PR TITLE
Remove multiple-statement sig support

### PR DESCRIPTION
**Pull request by elliottt**
Remove the ability to define signatures with multiple statements, and standardize on the chained syntax. Add a best-effort autocorrect to help with migration, that can be applied with the following:
```
srb tc --isolate-error=5069 -a
```

I would recommend reviewing by commit, as 2de8f5aa49f27385444e45265c41683b1a967747 involves a lot of whitespace changes.

### Motivation

Standardizing on a single syntax for signatures.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_KusYXSLLonddUz
→ https://go/builds/bui_KusYjN2ZCmX11Y

**Comment by elliottt** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_Kuvxx2GUMrUOU6
→ https://go/builds/bui_KuvxTbetcXUHbZ

File: test/testdata/resolver/non_builder_sig.rb [hunk 1]
```diff
@@ -1,10 +0,0 @@
-# typed: true
-extend T::Sig
-sig do
-  params(x: Integer)
-  returns(Integer)
-end
-def foo(x)
-  x
-end
-foo(3)
```
Comment (jez) [inline test/testdata/resolver/non_builder_sig.rb:10]: worth keeping as a test of the error?

File: resolver/type_syntax.cc [hunk 2]
```diff
@@ -105,44 +105,78 @@ bool TypeSyntax::isSig(core::Context ctx, const ast::Send &send) {
...
+
+        auto len = loc.endLoc - loc.beginLoc;
+        replacement.append(source.substr(loc.beginLoc, len));
+
+        first = false;
+    }
+
+    e.replaceWith("Use a chained sig builder", core::Loc{ctx.file, insseq->loc}, "{}", replacement);
+
+    return;
```
Comment (jez) [inline resolver/type_syntax.cc:151]:
```suggestion
```

**Review by jez — APPROVED** [review]

**Comment by jez** [comment]
What do you think about adding a cli test to test that the autocorrect works?

File: resolver/type_syntax.cc [hunk 3]
```diff
@@ -105,44 +105,78 @@ bool TypeSyntax::isSig(core::Context ctx, const ast::Send &send) {
...
-                }
-                send = ast::cast_tree<ast::Send>(insseq->expr);
-                if (!send) {
-                    return sig;
-                }
-                sends.emplace_back(send);
-            } else {
-                return sig;
+            if (auto e = ctx.beginError(sigSend.loc, core::errors::Resolver::MultipleStatementsInSig)) {
+                e.setHeader("Malformed `sig`: Signature blocks must contain a single statement");
```
[the remark this answers is not in the document]
Comment (elliottt) [reply resolver/type_syntax.cc:172]: I've made this change in a separate PR. I'll set you as the reviewer on that one.

**Comment by elliottt** [comment]
https://go/ps/pull/406381 is the Stripe migration PR.

**Comment by elliottt** [comment]
> What do you think about adding a cli test to test that the autocorrect works?

Sure, I'll add that tomorrow.

**Pull request merged**